cli/src/main/java/org/jboss/sbomer/cli/commands/processor/AbstractBaseProcessCommand.java

```diff
@@ -65,31 +70,18 @@ public Integer call() throws Exception {
         // Run the actual processing
         Bom processedBom = doProcess(bom);
 
-        // Create the new SBOM based on the old one with updated fields.
-        Sbom newSbom = processed(sbom, processedBom);
+        sbomerClient.updateSbom(String.valueOf(sbom.getId()), SbomUtils.toJsonNode(processedBom));
 
-        // Save the SBOM in the service
-        sbomerClient.save(newSbom);
+        log.info("SBOM with id '{}' updated!", sbom.getId());
 
         return CommandLine.ExitCode.OK;
     }
 
-    private Sbom processed(Sbom originalSbom, Bom bom) {
-        Sbom processed = new Sbom();
-        processed.setBuildId(originalSbom.getBuildId());
-        processed.setGenerator(originalSbom.getGenerator());
-        processed.setProcessor(getImplementationType());
-        processed.setType(originalSbom.getType());
-        processed.setSbom(SbomUtils.toJsonNode(bom));
-
-        return processed;
-    }
-
     protected abstract ProcessorImplementation getImplementationType();
 
     protected Bom doProcess(Bom bom) {
         log.info(
-                "Applying {} processing to the SBOM: {}",
+                "Applying {} processing to the SBOM: '{}'",
                 getImplementationType(),
                 bom.getMetadata().getComponent().getPurl());
```

Comment:
This log line might throw NPE, see LN 89, we need to handle the case where metadata or component are null

Reply:
Good catch!

Reply:
Fixed!